contracts/src/systems/resources/transfer_resources.cairo

```diff
@@ -0,0 +1,74 @@
+#[system]
+mod TransferResources {
+    use eternum::alias::ID;
+    use eternum::components::resources::Resource;
+    use eternum::components::owner::Owner;
+    use eternum::components::position::Position;
+    use eternum::components::config::WeightConfigImpl;
+
+    use dojo::world::Context;
+
+    use core::traits::Into;
+    use core::array::SpanTrait;
+
+    fn execute(ctx: Context, sending_entity_id: ID, receiving_entity_id: ID, mut resources: Span<(u8, u128)>) {
+
+        let sending_entity_owner = get!(ctx.world, sending_entity_id, Owner);
+        assert(sending_entity_owner.address == ctx.origin, 'not owner of entity id');
+        
+        // compare positions
+        let sending_entity_position = get!(ctx.world, sending_entity_id, Position);
+        let receiving_entity_position = get!(ctx.world, receiving_entity_id, Position);
+
+        assert(receiving_entity_position.x !=  0, 'entity position mismatch');
+        assert(receiving_entity_position.y != 0, 'entity position mismatch');
+
+        assert(receiving_entity_position.x == sending_entity_position.x, 'entity position mismatch');
+        assert(receiving_entity_position.y == sending_entity_position.y, 'entity position mismatch');
+       
+        // get receiving entity's total capacity
```

Comment:
this is good, one remark:
if the receiving entity does not have a capacity and is not movable, then it can receive any amount of resources.
The logic is that if no capacity/movable, then it must be something like a Realm or Hyperstructure. On which you can put unlimited amount of resources.

I think this would work for now, let's see in the future if there are flaws in this approach.

Reply:
ok i see, that's ok